handbrake: 1.2.2 -> 1.3.0

acowley opened the pull request:
This update is important as handbrake 1.2.2 no longer builds with
nixpkgs due to an unrelated update to pango-1.44.x as described
[here](https://github.com/HandBrake/HandBrake/issues/2226).

Fixes #73399

<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

acowley committed:
handbrake: 1.2.2 -> 1.3.0

This update is important as handbrake 1.2.2 no longer builds with
nixpkgs due to an unrelated update to pango-1.44.x as described
[here](https://github.com/HandBrake/HandBrake/issues/2226).

Fixes #73399

acowley commented:
Thanks to @jtojnar for showing the way forward with [this comment](https://github.com/NixOS/nixpkgs/issues/73399#issuecomment-565778648).

acowley commented on pkgs/applications/video/handbrake/default.nix:
```diff
@@ -96,6 +92,8 @@ stdenv.mkDerivation rec {
     substituteInPlace libhb/module.defs \
       --replace '$(CONTRIB.build/)include/libxml2' ${libxml2.dev}/include/libxml2
 
+    substituteInPlace gtk/configure.ac --replace 'HB_LIBS="-lhandbrake -lavformat -lavfilter -lavcodec -lavutil -ldav1d -lswresample -lpostproc -ldvdnav -ldvdread -lmp3lame -lvorbis -lvorbisenc -logg -lx264 -lswscale -ltheoraenc -ltheoradec -lvpx -lz -lbz2 -lbluray -lass -lfontconfig -lfreetype -lxml2 -ljansson -lopus -lspeex -llzma"' 'HB_LIBS="-lhandbrake -lavformat -lavfilter -lavcodec -lavutil ...
```
I'm happy to do something else. I didn't add to `NIX_LDFLAGS` because it's one specific part of the build that's not working for us: the GTk-using `ghb` executable. The library is mentioned for that build [here](https://github.com/HandBrake/HandBrake/blob/master/gtk/configure.ac#L256), but I'm not sure how our build is not working as expected, so I opted for something to minimize the patching. In that respect, I'd prefer `NIX_LDFLAGS`, too.

jtojnar replied:
That line is only a fallback branch. For us, it would have to be in the `*-*-linux*` branch. And apparently they fixed it sometime after the release:

https://github.com/HandBrake/HandBrake/commit/f7bc55eb2cf06759b2cbe339c858d1d6a6b2d831#diff-3bfc7b7ac6e2f746f2fa2173192d96fdR253

acowley commented on pkgs/applications/video/handbrake/default.nix:
```diff
@@ -118,6 +116,12 @@ stdenv.mkDerivation rec {
     cd build
   '';
 
+  dontUseNinjaBuild = true;
```
I didn't want to aggravate a potentially heated situation with upstream as described in the nixpkgs issue. I'll try another approach to see if you prefer it.

acowley commented:
I've done things differently to possibly better insulate our changes to the upstream build.

jtojnar commented on pkgs/applications/video/handbrake/default.nix:
```diff
@@ -109,9 +104,12 @@ stdenv.mkDerivation rec {
 
-  # NOTE: 2018-12-27: Check NixOS HandBrake test if changing
+  # NOTE 1: 2018-12-27: Check NixOS HandBrake test if changing
+  # NOTE 2: We have to patch libdl in to the ghb linking step because it is
```
Could you move the comment just before the `-ldl`? Also mention that it will be fixed in next release.

acowley replied:
I removed my comment. I'm not familiar with plans for fixing before the next release.

jtojnar replied:
It was fixed in https://github.com/HandBrake/HandBrake/commit/f7bc55eb2cf06759b2cbe339c858d1d6a6b2d831#diff-3bfc7b7ac6e2f746f2fa2173192d96fdR253.

acowley replied:
Ah, thank you!

jtojnar commented on pkgs/applications/video/handbrake/default.nix:
```diff
@@ -110,8 +105,10 @@ stdenv.mkDerivation rec {
   ];
 
   # NOTE: 2018-12-27: Check NixOS HandBrake test if changing
+  # NOTE: The -ldl flag was fixed upstream for a release after 1.3.0
```
Could you move it inside the list, just before the relevant line?

acowley replied:
Done.

jtojnar commented on pkgs/applications/video/handbrake/default.nix:
```diff
@@ -99,6 +93,7 @@ stdenv.mkDerivation rec {
     # Force using nixpkgs dependencies
     sed -i '/MODULES += contrib/d' make/include/main.defs
     sed -i '/PKG_CONFIG_PATH=/d' gtk/module.rules
+    sed -i 's/^[[:space:]]*\(meson\|ninja\)[[:space:]]*= ToolProbe.*$//g' make/configure.py
```
Could this be made for cmake as well?

jtojnar replied:
Also possibly for nasm.

acowley replied:
Removing nasm followed the same pattern. cmake involved a larger edit that I'm not sure will be as well received. Let me know.

acowley committed:
handbrake: remove nasm and cmake dependencies

These were included by the handbrake build because they are used by external dependencies. In nixpkgs, we build those dependencies (e.g. codecs) separately, and so excise the check for these tools from the handbrake build.

jtojnar reviewed (approved):
LGTM, program starts.

Lassulus reviewed (approved):
tested with nix-review, seems to work, weird behavior when I try to open files
